### ISSUE TITLE
Strange framerate drop in Shin Megami Tensei IV

### ISSUE THREAD

**Issue by LOuroboros** [citra-emu/citra#5051]
This is an issue I found back when I first played the game, and I absolutely forgot to report it, so I'm doing it now.

Tested on Citra Canary 1529.
Basically, the framerate goes down when I select certain side quests. It's super weird, so I'm assuming it must be some kind of issue with Citra...?

Here's a quick video showing it off: https://streamable.com/8jtr6
Here's my citra_log.txt file: https://pastebin.com/EpsBu1ZK

**PC Specs**
**CPU**: i3 2120 _(2c/4t running at 3.29-3.30 GHz)_
**RAM**: 4 GB _(Single Channel)_
**GPU**: NVIDIA GeForce GT 1030 _(Graphics Driver v441.66)_
**OS**: Windows 8.1 Embedded Industry Pro _(64 Bits)_

**Comment by viocar** [comment]
I don't think this is necessarily Citra's fault. I've seen other games lag in similar situations - the loops that figure out all the text to render as well as any important quest flags that may change the text can be surprisingly CPU-intensive.

**Comment by tywald** [comment]
Hm, issue reports should be based on Nightly builds. If it's a problem on Canary a comment should be put for the responsible PR(you can test appveyor builds to find out which PR is causing problems).

**Comment by LOuroboros** [comment]
> I don't think this is necessarily Citra's fault. I've seen other games lag in similar situations - the loops that figure out all the text to render as well as any important quest flags that may change the text can be surprisingly CPU-intensive.

Hm. Alright then. Should I close the issue, or is there a way for a user like me to verify if this is actually a problem with a lack of power?

> Hm, issue reports should be based on Nightly builds. If it's a problem on Canary a comment should be put for the responsible PR(you can test appveyor builds to find out which PR is causing problems).

When creating an issue, the instructions in the comment box don't really mention that you must use a specific version of the emulator to report issues with it, though it does say that you must clarify which version are you using.
Sorry I guess, I'll keep that in mind for next time.

**Comment by LOuroboros** [comment]
This is kind of strange, but when I reopened the emulator a few days after posting this report, the issue seemed to vanish.

Could it be that Citra eats more CPU resources the longer it stays opened, and that could have caused the problem to begin with?

**Comment by LOuroboros** [comment]
I spotted the error yet again with a new side quest that I got.
When I checked the Task Manager, I noticed that Citra (Nightly 1399) was using 850 to 900 MBs of RAM.
I saved the game and restarted Citra, and the usage of RAM was of 550 to 600 MB.
Is that normal?

**Comment by B3n30** [comment]
Sadly yes. We cache all the used textures and at the current stage we never clear the cache, so after long playing citra there will be most likely performance issues. Closing citra and open it again should bring back the performance.

In the future we need a way to remove old unused textures from the cache

**Comment by LOuroboros** [comment]
> Closing citra and open it again should bring back the performance.

Just to clarify, the issue while reading the summary of certain side quests remains, even if you restart the emulator.

I would like to find out if the problem comes from a lack of power as viocar implied, or if the issue comes from a lacking in Citra's emulation of the 3DS.

Here's another test after having just restarted Citra.
https://streamable.com/6uytu

Here's my current citra_log: https://pastebin.com/yUsLmcz7

**Comment by BreadFish64** [comment]
It seems like the issue is due to the graphics driver automatically recompiling a shader based on a changed uniform.

**Comment by LOuroboros** [comment]
> 
> 
> It seems like the issue is due to the graphics driver automatically recompiling a shader based on a changed uniform.

Oh. NVIDIA's GPU Drivers' Control Panel do grant the user access to a Shader Cache feature. Could that be interfering with Citra's or something?

**Comment by LOuroboros** [comment]
This is still happening in Canary 1970, by the way.
Selecting the side quest "Serial Kidnapping + Arson" makes the framerate drop to 25 FPS, where as when I select selecting the quest right after it on my list, "The Great Drunkenness", it goes back to 30.

**Comment by GPUCode** [comment]
Has this been fixed?
